packages/synapse-react-client/src/components/styled/StyledPopover.tsx

```diff
@@ -0,0 +1,122 @@
+import LightTooltip from '@/components/styled/LightTooltip'
+import { spreadSx } from '@/theme/spreadSx'
+import {
+  Box,
+  Button,
+  ButtonProps,
+  SxProps,
+  tooltipClasses,
+  TooltipProps,
+} from '@mui/material'
+import { atom, useAtom } from 'jotai'
+import { PropsWithChildren, ReactNode, useId } from 'react'
+
+export type StyledPopoverProps = PropsWithChildren<{
+  popoverContent: React.ReactNode
+  sx?: TooltipProps['sx']
+  placement?: TooltipProps['placement']
+  showCloseButton?: boolean
+  actionButton?: {
+    content: ReactNode
+    color?: ButtonProps['color']
+    variant?: ButtonProps['variant']
+    onClick: () => void
+    closePopoverOnClick?: boolean
+  }
+  maxWidth?: string
+  minWidth?: string
+  containerSx?: SxProps
+}>
+
+const buttonBoxSx: SxProps = {
+  mt: 2,
+  display: 'flex',
+  justifyContent: 'space-between',
+  flexDirection: 'row-reverse',
+  '> button': {
+    flexGrow: 1,
+    maxWidth: '45%',
+  },
+}
+
+// Register a global atom to track which popover is open, to ensure only one is shown at any given time
+const openMarkdownPopoverAtom = atom<string | null>(null)
+
+export function StyledPopover({
```

Comment:
Identical to old MarkdownPopover except renders an arbitrary ReactNode. MarkdownPopover now uses this.